File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> sk;
        for(char ch:s){
            switch(ch){
                case '{':
                case '(':
                case '[':
                    sk.push(ch);
                    break;
                case '}':
                    if(sk.empty())
                        return false;
                    if(sk.top()=='{')
                        sk.pop();
                    else 
                        return false;
                    break;
                case ')':
                    if(sk.empty())
                        return false;
                    if(sk.top()=='(')
                        sk.pop();
                    else 
                        return false;
                    break;
                case ']':
                    if(sk.empty())
                        return false;
                    if(sk.top()=='[')
                        sk.pop();
                    else 
                        return false;
                    break;
            }
        }
        if(sk.empty())
            return true;
        return false;
    }
};

File: Data Structures & Algorithms/baseball-game/submission-1.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> records = new Stack<>();
        for(String op: operations){
            if(op.equals("+")){
                Integer second = records.pop();
                Integer first = records.peek();
                records.push(second);
                records.push(first+second);
            } else if(op.equals("D")) {
                records.add(2*records.peek());
            } else if(op.equals("C")) {
                records.pop();
            } else {
                records.push(Integer.parseInt(op));
            }
        }
        Integer sum = 0;
        while(!records.empty())
            sum += records.pop();
        return sum;
    }
}